presto-common/src/main/java/com/facebook/presto/common/type/LongEnumType.java

```diff
@@ -72,12 +81,15 @@ public String getDisplayName()
     public static class LongEnumMap
     {
         private final Map<String, Long> enumMap;
+        private Map<Long, String> flippedEnumMap;
+        private final AtomicBoolean isFlippedEnumComputed = new AtomicBoolean();
```

Comment:
You can simply use the existing guava feature.
```Java
private Supplier<Map<Long, String>> flippedEnumMap = Supplier.memorize(this::getKeyForValue);
```

according to its Javadoc
> Returns a supplier which caches the instance retrieved during the first call to {@code get()} and returns that value on subsequent calls to {@code get()}

Reply:
same problem there – guava isn't a runtime dependency of presto-common